### ISSUE TITLE
Unable to see roles of all system users as a global super admin on a team

### ISSUE THREAD

**Issue by sts-ryan-holton** [spatie/laravel-permission#2299]
**Describe the bug**

I'm using the teams feature of this package. In my context it's a company so I'll refer to it as that throughout. I have a `Company` model and a `User` can only ever have one company, then, a `Company` can have as many roles as is needed.

I have a global Super admin that isn't assigned to any company.

As a super admin, when I log in and go to my user management page I need to be able to see the roles assigned to each user, not just the roles for me who I'm logged in as.

I believe this is related to the `setPermissionsTeamId` function which can only ever hold one company id, but as a super admin I need to be able to see everyone's.

**Versions**

- spatie/laravel-permission package version: 5.7.0
- illuminate/framework package 9.47.0

PHP version: 8.1.0

Database version: MySQL

 **Additional context**

Here's how I'm getting a list of users:

```php
/**
 * Display a listing of the resource.
 *
 * @return \Illuminate\Http\Response
 */
public function index(Request $request)
{
    $this->authorize('viewAny', User::class);

    $validator = Validator::make($request->all(), [
        'perPage' => [
            'numeric',
            new ValidPaginationPerPage
        ]
    ]);

    if ($validator->fails()) {
        return response()->json([
            'message' => 'One or more fields has been missed or is invalid.',
            'errors' => $validator->messages(),
        ], 400);
    }

    $users = User::with('roles')->paginate($request->input('perPage', 10));
    $users->onEachSide(1);

    if (!$users || count($users) <= 0) {
        return response()->json([
            'message' => 'No users found'
        ], 404);
    }

    return response()->json([
        'users' => $users
    ], 200);
}
```

Here's what my users view looks like whilst logged in as a super admin:

![list-roles](https://user-images.githubusercontent.com/25177538/212371659-b96f9013-48f5-4959-8b0d-467bb51519c9.png)

And here's my table of roles:

<img width="732" alt="table-roles" src="https://user-images.githubusercontent.com/25177538/212371716-21b14d16-5152-42e5-a347-b2d9b7ef506c.png">

To summarise, a super admin should be able to see all roles of all users regardless of company.

**Comment by parallels999** [comment]
>  I've seen you post several issues which are support questions. Please keep in mind the issue templates when submitting issues.

https://github.com/laravel/fortify/issues/380#issuecomment-1091554521

I see you do the same everywhere and you don't understand it

**Comment by sts-ryan-holton** [comment]
@parallels999 

> It is not a bug or issue, it is a expected behaivor

This isn't an acceptable answer given the number of issues related to the Teams feature.

Having a Teams feature implies that I can have more than one team, and as such I should be able to see other teams/companies regardless of who I'm logged in as if my role is a global super admin.

Linking to Fortify isn't related at all to this package.

Please investigate further.

**Comment by parallels999** [comment]
It is still expected behavior.

> Having a Teams feature implies that I can have more than one team, and as such I should be able to see other teams/companies regardless of who I'm logged in as if my role is a global super admin.

if you read other issues/discussions/pull requests about this you would know how to do it, I will no longer perform the searches for you, never even a "thank you" you return

> This isn't an acceptable answer given the number of issues related to the Teams feature.

Feel free to make a pull request to make it better

**Comment by sts-ryan-holton** [comment]
If my described issue has already been discussed, please link me to it - this would be helpful.

**Comment by parallels999** [comment]
> I will no longer perform the searches for you, never even a "thank you" you return

There you can search, or you can wait for someone else to do the searching for you, and don't forget to thank whoever does it, this is not a paid job, those who help use their personal time to do it,
https://github.com/spatie/laravel-permission/issues?q=is%3Aopen
![image](https://user-images.githubusercontent.com/109294935/212381220-4e262c9f-1fce-49f9-a1b1-40aa5ef461fd.png)
have a good day

**Comment by erikn69** [comment]
Easy
- https://github.com/spatie/laravel-permission/pull/1948

**Comment by drbyte** [comment]
Scoping as per the examples erikn69 linked to should resolve your situation.

**Comment by andredsnogueira** [comment]
Hmm, strange that I cannot make it work using the scopeUser. Does it work still? I am using latest version of this package and latest Laravel version.
